byway/src/main/java/com/google/sps/data/Trip.java

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A class to make a Trip type, containing a specific id, destinations, interests and routes to be
+ * used when updating a trip.
+ */
+public final class Trip {
+
+  public static final String DATASTORE_ENTITY_KIND = "Trip";
+
+  private final String keyString;
+  private final String start;
+  private final ArrayList<String> destinations;
+  private final ArrayList<String> interests;
+  private final ArrayList<String> route;
+
+  public Trip(
+      String keyString,
+      String start,
+      Collection<String> destinations,
+      Collection<String> interests,
+      Collection<String> route) {
+    this.keyString = checkNotNull(keyString, "keyString");
+    this.start = checkNotNull(start, "start");
+    checkNotNull(destinations, "destinations");
+    checkNotNull(interests, "interests");
+    checkNotNull(route, "route");
+    this.destinations = new ArrayList<String>(destinations);
+    this.interests = new ArrayList<String>(interests);
+    this.route = new ArrayList<String>(route);
+  }
+
+  public List<String> getInterests() {
```

Comment:
Can you add comments to these methods to describe what they are? Not just like "the user's interests", but rather it would be good to include information about the provenance of the data and what format it is in. E.g. "Interests the user selected for the trip, as free-form, user-supplied text."

Applies to all getters for both data classes here.

Reply:
Updated